app/src/lib/stores/app-store.ts

```diff
@@ -2843,76 +2847,88 @@ export class AppStore extends TypedBaseStore<IAppState> {
     const { changesState, branchesState } = this.repositoryStateCache.get(
       repository
     )
-    const { tip } = branchesState
-    const currentBranch = tip.kind === TipState.Valid ? tip.branch : null
-    const hasChanges = changesState.workingDirectory.files.length > 0
-
-    if (
-      enableStashing() &&
-      hasChanges &&
-      currentBranch !== null &&
-      uncommittedChangesStrategy ===
-        UncommittedChangesStrategy.askForConfirmation
-    ) {
-      this._showPopup({
-        type: PopupType.StashAndSwitchBranch,
-        branchToCheckout: foundBranch,
-        repository,
-      })
 
-      return repository
-    }
+    let createdStashEntry = null
```

Comment:
I like that idea. Unfortunately, we cannot initialize it with the value of `uncommittedChangesStrategy.transientStashEntry` because the `transientStashEntry` property is when `uncommittedChangesStrategy.kind` is `UncommittedChangesStrategyKind.moveToNewBranch`.

I'm 👍 on renaming this variable though. If I do so, L2926 will look like the following code snippet:

```ts
stashToPop = stashToPop || uncommittedChangesStrategy.transientStashEntry
      if (stashToPop !== null) {
        await gitStore.performFailableOperation(() => {
          // we have to use the `!` operator since TS thinks this object can be null...
          return popStashEntry(repository, stashToPop!.stashSha)
        })
      }
```

If you're 👍 on that change, I can go ahead and do it, otherwise, maybe I can choose another name that won't require us to use the non-null assertion.

Reply:
I think the ideal refactor here would be to encapsulate the following code into its own private function on AppStore that returns `Promise<StashEntry | null>` so we can assign that result directly to `stashToPop`.

https://github.com/desktop/desktop/blob/49d5d54c9096b7a8b9d9b0cc93664033f11c5a04/app/src/lib/stores/app-store.ts#L2885-L2916

Will file an issue to track this refactor.

Reply:
captured in https://github.com/desktop/desktop/issues/7609